spec/reek/examiner_spec.rb

```diff
@@ -108,5 +108,42 @@
         expect(examiner.smells).to eq([])
       end
     end
+
+    context 'with an incomprehensible source that causes Reek to crash' do
+      let(:subject) do
+        examiner = described_class.new 'class C; def does_crash_reek; end; end'
+        allow(examiner).to receive(:smell_repository) do
```

Comment:
> Are you suggesting to update the Examiner constructor to be like this:

We probably need to inject the class instead of an instance, but yes, something like that.

> > or stubbing SmellRepository.new directly?
> I can't see the benefit. For me this would be even worse since it is more opaque - you're stubbing something that is not even called directly thus making it harder to build up a mental image. Why do you think this is better?

I think `SmellRepository.new` is actually called directly in `Examiner`. But to answer more broadly, I have found again and again that stubbing the object under test is really the most dangerous option, since it tends to hide bugs in the implementation of the object itself.